cypress/e2e/projectSettings.cy.ts

```diff
@@ -0,0 +1,393 @@
+            cy.get(projectSettingsTransferProjectUserForm.query).find('input').type(userEmail);
+            cy.get(popupAttribute).contains('div', userEmail).click();
+
+            cy.get(projectSettingsCancelTransferProjectButton.query).should('not.be.disabled');
+            cy.get(projectSettingsConfirmTransferProjectButton.query).should('not.be.disabled');
+
+            cy.get(projectSettingsCancelTransferProjectButton.query).click();
+            cy.get(projectSettingsConfirmTransferProjectButton.query).should('not.exist');
+        });
+
+        it('confirm transferring to other user', () => {
+            cy.get(projectSettingsTransferProjectButton.query).click();
+
+            cy.get(projectSettingsCancelTransferProjectButton.query).should('not.be.disabled');
+            cy.get(projectSettingsConfirmTransferProjectButton.query).should('be.disabled');
+
+            cy.get(projectSettingsTransferProjectKeyInput.query).type(id);
+
+            cy.get(projectSettingsCancelTransferProjectButton.query).should('not.be.disabled');
+            cy.get(projectSettingsConfirmTransferProjectButton.query).should('be.disabled');
+
+            cy.get(projectSettingsTransferProjectUserForm.query).find('button').click();
+            cy.get(projectSettingsTransferProjectUserForm.query).find('input').type(userEmail);
+            cy.get(popupAttribute).contains('div', userEmail).click();
+
+            cy.get(projectSettingsCancelTransferProjectButton.query).should('not.be.disabled');
+            cy.get(projectSettingsConfirmTransferProjectButton.query).should('not.be.disabled');
+
+            cy.get(projectSettingsConfirmTransferProjectButton.query).click();
+        });
+    });
+
+    describe('check routing', () => {
+        describe('user is owner', () => {
+            beforeEach(() => {
+                cy.clearCookies();
+                cy.signInViaEmail({ email: userEmail, password: userPassword });
+            });
+
+            it('going via tab from project page', () => {
+                cy.visit(exactUrl(routes.project(id)));
+
+                cy.get(projectPageTabs.query)
+                    .find(`[href="${routes.projectSettings(id)}"]`)
+                    .click()
+                    .wait(500);
+                cy.url().should('equal', exactUrl(routes.projectSettings(id)));
+            });
+
+            it('going via exact url from project page', () => {
+                cy.visit(exactUrl(routes.projectSettings(id)));
+                cy.url().should('equal', exactUrl(routes.projectSettings(id)));
+            });
+        });
+
+        describe('admin is NOT owner', () => {
+            it('going via tab from project page', () => {
+                cy.visit(exactUrl(routes.project(id)));
+
+                cy.get(projectPageTabs.query)
+                    .find(`[href="${routes.projectSettings(id)}"]`)
+                    .click()
+                    .wait(500);
```

Comment:
Please try to find a way not to wait constant time. It's better to wait for an element/page load. Sometimes it's not possible and there you should leave a comment explaining the details.